FreeRTOS-Plus/Demo/MQTT_Mutual_Auth_Qemu_mps2/mbedtls_freertos_port.c

```diff
@@ -0,0 +1,260 @@
+    configASSERT( pMutex->mutexHandle != NULL );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Frees a mutex.
+ *
+ * @param[in] pMutex mbedtls mutex handle.
+ *
+ * @note This function is an empty stub as nothing needs to be done to free
+ * a statically allocated FreeRTOS mutex.
+ */
+void mbedtls_platform_mutex_free( mbedtls_threading_mutex_t * pMutex )
+{
+    /* Nothing needs to be done to free a statically-allocated FreeRTOS mutex. */
+    ( void ) pMutex;
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Function to lock a mutex.
+ *
+ * @param[in] pMutex mbedtls mutex handle.
+ *
+ * @return 0 (success) is always returned as any other failure is asserted.
+ */
+int mbedtls_platform_mutex_lock( mbedtls_threading_mutex_t * pMutex )
+{
+    BaseType_t mutexStatus = 0;
+
+    configASSERT( pMutex != NULL );
+
+    /* mutexStatus is not used if asserts are disabled. */
+    ( void ) mutexStatus;
+
+    /* This function should never fail if the mutex is initialized. */
+    mutexStatus = xSemaphoreTake( pMutex->mutexHandle, portMAX_DELAY );
+    configASSERT( mutexStatus == pdTRUE );
+
+    return 0;
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Function to unlock a mutex.
+ *
+ * @param[in] pMutex mbedtls mutex handle.
+ *
+ * @return 0 is always returned as any other failure is asserted.
+ */
+int mbedtls_platform_mutex_unlock( mbedtls_threading_mutex_t * pMutex )
+{
+    BaseType_t mutexStatus = 0;
+
+    configASSERT( pMutex != NULL );
+    /* mutexStatus is not used if asserts are disabled. */
+    ( void ) mutexStatus;
+
+    /* This function should never fail if the mutex is initialized. */
+    mutexStatus = xSemaphoreGive( pMutex->mutexHandle );
+    configASSERT( mutexStatus == pdTRUE );
+
+    return 0;
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Function to generate a random number. Updated to use rand() for Qemu devices 
+ *
+ * @param[in] data Callback context.
+ * @param[out] output The address of the buffer that receives the random number.
+ * @param[in] len Maximum size of the random number to be generated.
+ * @param[out] olen The size, in bytes, of the #output buffer.
+ *
+ * @return 0 if no critical failures occurred,
+ * MBEDTLS_ERR_ENTROPY_SOURCE_FAILED otherwise.
+ */
+int mbedtls_platform_entropy_poll( void * data,
+                                   unsigned char * output,
+                                   size_t len,
+                                   size_t * olen )
+{
+    static BaseType_t RNGIsSeeded = pdFALSE;
+    int status = 0;
+
+    if (RNGIsSeeded == pdFALSE)
+    {
+        srand((unsigned int) time (NULL));
+    }
+    configASSERT( output != NULL );
+    configASSERT( olen != NULL );
+
+    /* Context is not used by this function. */
+    ( void ) data;
+
+    /* TLS requires a secure random number generator; use the RNG provided
```

Comment:
Update this comment - this is no longer windows. might want to add that this is pseudo-random and not a good source of entropy so people should update it if they plan to build a project off this. prob add it to the readme as well.